src/compiler/transformers/ts.ts

```diff
@@ -1134,26 +1147,36 @@ namespace ts {
          *
          * @param node The class node.
          * @param isStatic A value indicating whether to get properties from the static or instance side of the class.
-         */
-        function getInitializedProperties(node: ClassExpression | ClassDeclaration, isStatic: boolean): ReadonlyArray<PropertyDeclaration> {
-            return filter(node.members, isStatic ? isStaticInitializedProperty : isInstanceInitializedProperty);
-        }
-
-        /**
-         * Gets a value indicating whether a class element is a static property declaration with an initializer.
-         *
-         * @param member The class element node.
-         */
-        function isStaticInitializedProperty(member: ClassElement): member is PropertyDeclaration {
-            return isInitializedProperty(member, /*isStatic*/ true);
+         * @param hoistedComputedNames An output value to push the any expressions which need hoisting
+         * @param decoratedMembers An output value which holds all members with the appropriate staticness and have decorators
+         */
+        function getInitializedProperties(node: ClassExpression | ClassDeclaration, isStatic: boolean, hoistedComputedNames: Push<BinaryExpression>, decoratedMembers: Push<ClassElement>): ReadonlyArray<PropertyDeclaration> {
+            if (!node.members) return;
+            const result: PropertyDeclaration[] = [];
+            for (let member of node.members) {
+                if (!isStatic && hasComputedNameWhichRequiresHoisting(member)) {

```

Comment:
It feels like this function now has too many responsibilities. I'd much rather keep the `hoistedComputedNames` and `decoratedMembers` parts separate.